tests/test_excise_management/tests/test_action_server.py

```diff
@@ -0,0 +1,87 @@
+            "Changing the excise category should remove the old category purchase taxes")
+        self.assertTrue(product_form.x_excise_category.x_sales_tax_id in product_template.taxes_id,
+            "Changing the excise category should add the new category sales taxes")
+        self.assertTrue(product_form.x_excise_category.x_purchase_tax_id in product_template.supplier_taxes_id,
+            "Changing the excise category should add the new category purchase taxes")
+
+    def test_create_excise_tax_server_action(self):
+        excise_category = self.env.ref("excise_management.x_excise_category_S001")
+        self.assertTrue(excise_category.x_sales_tax_id,
+            "Creating an excise category should give it sales taxes automatically")
+        self.assertTrue(excise_category.x_sales_tax_id.formula == 'quantity * product.x_excise_amount',
+            "The sales tax of the excise category should have its formula equal to 'quantity * product.x_excise_amount'")
+        self.assertTrue(excise_category.x_sales_tax_id.type_tax_use == 'sale',
+            "The sales tax of the excise category should have its type equal to 'sale'")
+        self.assertTrue(excise_category.x_sales_tax_id.tax_group_id == self.env.ref('excise_management.excises_tax_group'),
+            "The sales tax of the excise category should have its tax group set to the excises tax group")
+        self.assertTrue(excise_category.x_purchase_tax_id,
+            "Creating an excise category should give it purchase taxes automatically")
+        self.assertTrue(excise_category.x_purchase_tax_id.formula == 'quantity * product.x_excise_amount',
+            "The purchase tax of the excise category should have its formula equal to 'quantity * product.x_excise_amount'")
+        self.assertTrue(excise_category.x_purchase_tax_id.type_tax_use == 'purchase',
+            "The purchase tax of the excise category should have its type equal to 'purchase'")
+        self.assertTrue(excise_category.x_purchase_tax_id.tax_group_id == self.env.ref('excise_management.excises_tax_group'),
+            "The purchase tax of the excise category should have its tax group set to the excises tax group")
+        fiscal_positions = self.env['account.fiscal.position'].search([('x_is_fiscal_deposit', '=', False)])
+        if fiscal_positions:
+            self.assertTrue(excise_category.x_sales_tax_id in fiscal_positions.tax_ids,
+                "Fiscal positions that are not deposit should contains the sales taxes of the current category")
+            self.assertTrue(excise_category.x_purchase_tax_id in fiscal_positions.tax_ids,
+                "Fiscal positions that are not deposit should contains the purchase taxes of the current category")
+
+    def test_add_excise_taxes_fiscal_position_server_action(self):
+        excise_taxes = self.env['account.tax'].search([('x_is_excise', '=', True)])
+        self.assertEqual(self.fiscal_position.tax_ids, excise_taxes,
```

Comment:
You may have other taxes, right?
```suggestion
        self.assertEqual(excise_taxes, self.fiscal_position.tax_ids,
```

Reply:
I meant assertIn

Reply:
I really thought there was a hidden meaning in the assertEqual like acting as "in" so the order matters. Will correct it right away !

Reply:
Server action works as intended (if there is already a tax it does not remove it and add the new ones that are excise). It seems the in is not working properly to test if an array of taxes is in another array of taxes, so as a way around i just made a for loop on the taxes.